pysolr.py

```diff
@@ -580,7 +580,7 @@ def _from_python(self, value):
         """
         if hasattr(value, 'strftime'):
             if hasattr(value, 'hour'):
-                value = "%sZ" % value.isoformat()
+                value = "%sZ" % ( value.strftime('%Y-%m-%dT%H:%M:%S.%f') if value.tzinfo else value.isoformat() )
```

Comment:
This should be an `if` block for clarity since that's a pretty large line.

Also, as written above it looks like this will be incorrect for any datetime which isn't already in UTC since the date is being specified as UTC (`Z`). I think by now every version of Python which we support has the [`utcoffset`](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcoffset) method so this might be as simple as `value - value.utcoffset()` but we really should have some tests covering this.